langlib/langlib-test/src/test/resources/test-src/regexp_negative_test.bal

```diff
@@ -101,3 +102,52 @@ public function testInvalidRegexpPatternSyntax5() returns error? {
     string:RegExp x = re `(?xsmi:[]\P{sc=Braille})`;
     _ = check (trap x.matchGroupsAt(":*A*a"));
 }
+
+public function testNegativeDuplicateFlags1() returns error? {
+    anydata|error result = trap re `${"(?i-i:ABC))"}`;
+    check assertEqualMessage(result, "Invalid insertion in regular expression: duplicate flag 'i'");
+}
+
+public function testNegativeDuplicateFlags2() returns error? {
+    anydata|error result = trap re `ABC${"(?ims-xi:ABC))"}(?-:)`;
+    check assertEqualMessage(result, "Invalid insertion in regular expression: duplicate flag 'i'");
+}
+
+public function testNegativeDuplicateFlags3() returns error? {
+    anydata|error result = regexp:fromString("(?ms-ims:ABC))");
+    check assertEqualMessage(result, "Failed to parse regular expression: duplicate flag 'm'");
+}
+
+public function testNegativeDuplicateFlags4() returns error? {
+    anydata|error result = regexp:fromString("(?imi:(ABC))");
+    check assertEqualMessage(result, "Failed to parse regular expression: duplicate flag 'i'");
+}
+
+public function testNegativeDuplicateFlags5() returns error? {
+    anydata|error result = regexp:fromString("(?-imi:(ABC))");
+    check assertEqualMessage(result, "Failed to parse regular expression: duplicate flag 'i'");
+}
+
+public function testNegativeInvalidFlags1() returns error? {
+    anydata|error result = trap re `${"(?i-mk:ABC))"}`;
+    check assertEqualMessage(result, "Invalid insertion in regular expression: invalid flag in regular expression");
+}
+
+public function testNegativeInvalidFlags2() returns error? {
+    anydata|error result = trap re `ABC${"(?lms-xi:ABC))"}(?-:)`;
+    check assertEqualMessage(result, "Invalid insertion in regular expression: invalid flag in regular expression");
```

Comment:
Small suggestion: Shall we add a one test that include both errors.
anydata|error result = trap re `${"(?ii-mk:ABC))"}`;